File: src/main/java/io/junseok/mealmate/domain/restaurantmenu/entity/RestaurantMenu.java
package io.junseok.mealmate.domain.restaurantmenu.entity;

import io.junseok.mealmate.domain.restaurant.entity.Restaurant;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "restaurant_menu")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Builder
public class RestaurantMenu {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "restaurant_menu_id")
    private Long restaurantMenuId;

    @Column(name = "menu_name")
    private String menuName;

    @Column(name = "menu_price")
    private String menuPrice;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "restaurant_id")
    private Restaurant restaurant;
}

File: src/main/java/io/junseok/mealmate/domain/restaurantmenu/entity/RestaurantMenu.kt
package io.junseok.mealmate.domain.restaurantmenu.entity

import io.junseok.mealmate.domain.restaurant.entity.Restaurant
import javax.persistence.*

@Entity
@Table(name = "restaurant_menu")
class RestaurantMenu(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "restaurant_menu_id")
    val restaurantMenuId: Long? = null,

    @Column(name = "menu_name")
    var menuName: String,

    @Column(name = "menu_price")
    var menuPrice: String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "restaurant_id")
    val restaurant: Restaurant
)
